app/src/main/java/ingsw/group1/findmyphone/Manager.java

```diff
@@ -4,149 +4,147 @@
 import android.content.Intent;
 import android.util.Log;
 
+import ingsw.group1.findmyphone.activity.ActivityConstantsUtils;
+import ingsw.group1.findmyphone.alarm.AlarmManager;
+import ingsw.group1.findmyphone.location.CommandResponseLocation;
+import ingsw.group1.findmyphone.location.LocationManager;
 import ingsw.group1.msglibrary.ReceivedMessageListener;
 import ingsw.group1.msglibrary.SMSManager;
 import ingsw.group1.msglibrary.SMSMessage;
 import ingsw.group1.msglibrary.SMSPeer;
 
-/***
+/**
+ * This is the main manager
+ *
  * @author Pardeep
- * Class
+ * @author Giorgia Bortoletti (refactoring)
  */
-
 public class Manager {
-    private static Context currentContext;
-    AlarmManager alarmManager = new AlarmManager();
-    LocationManager locationManager=new LocationManager();
-    SMSManager smsManager;
-    Constants constants=new Constants();
-
     private static final String MANAGER_TAG = "Manager";
-    private  SendResponseSms sendResponseSms;
+    private static Context currentContext;
 
-    public Manager(Context context){
-        currentContext=context.getApplicationContext();
-        smsManager=SMSManager.getInstance(Manager.currentContext);
+    private AlarmManager alarmManager;
+    private LocationManager locationManager;
+    private SMSManager smsManager;
+    private CommandResponseLocation sendResponseSms;
 
+    /**
+     * Constructor
+     *
+     * @param context application context
+     */
+    public Manager(Context context) {
+        currentContext = context.getApplicationContext();
+        locationManager = new LocationManager();
```

Comment:
Careful, you have a constructor that sets a static field, wasn't an issue before because `currentContext` wasn't static, you can leave it like this because the result remains correct, but only because we are calling `getApplicationContext()`, which will always be the same as long as the app runs.